MUD-287: Refunds for StakePayment [DO NOT MERGE!]

Whytecrowe opened the pull request.

RyanCross commented on contracts/registrar/Registrar.sol:
```diff
@@ -25,6 +25,14 @@ contract Registrar is
     uint256 royaltyAmount;
     uint256 parentId;
     address subdomainContract;
+    // This is the folder group the domain belongs to
```
I'm assuming this is for IPFS but it would be nice to give some context to the concept of a folder group either here or in the readme somewhere.

Remscar replied:
Will add a good documentation pass to the backlog

RyanCross commented on contracts/registrar/ZNSHub.sol:
```diff
@@ -32,6 +32,20 @@ contract ZNSHub is
+    address controller,
+    string metadataUri,
+    uint256 royaltyAmount,
+    uint256 groupId,
```
nit: `folderGroupId` & `folderGroupIndex`

Remscar replied:
We had a lengthy discussion about this weeks ago and decided to go with just `groupId` to keep it generic

RyanCross commented on contracts/registrar/ZNSHub.sol:
```diff
@@ -53,6 +67,12 @@ contract ZNSHub is
     address childRegistrar
   );
 
+  event EEDomainGroupUpdatedV1(
```
This domain group concept, is it the idea that every registrar will have its own folderGroup? Will the fileGroupIndexes on a domain ever need to change when a group is updated?

Remscar replied:
Not every registrar has it's own group. Sets of domains in a registrar may be in a group together. Multiple groups can exist on a registrar.

Remscar replied:
The indexes of domains in a group will never change.

RyanCross commented on contracts/registrar/ZNSHub.sol:
```diff
@@ -53,6 +67,12 @@ contract ZNSHub is
+  event EEDomainGroupUpdatedV1(
+    address parentRegistrar,
+    uint256 folderGroupId,
+    string baseUri
```
base`What`Uri?

Remscar replied:
for the metadata.

RyanCross commented on scripts/v1.1/upgrade-hub.ts:
```diff
@@ -2,7 +2,9 @@ import * as hre from "hardhat";
-const hubAddress = "0x3F0d0a0051D1E600B3f6B35a07ae7A64eD1A10Ca";
+//const hubAddress = "0x3F0d0a0051D1E600B3f6B35a07ae7A64eD1A10Ca";
+// rinkeby hub
+const hubAddress = "0x90098737eB7C3e73854daF1Da20dFf90d521929a";
```
Should update this value in https://github.com/zer0-os/zer0-contracts and then consume the package.

Remscar replied:
In a different PR we can do this

RyanCross reviewed (commented):
Looks good. I feel like I have an understanding of the changes needed to make the necessary updates for the data store.

I am worried that non passive changes in smart contracts (new events, the addition of a new event hub address) could easily slip by and start causing problems for the data store.  Would you be open to adding a PR template in smart contract repos that informs the user an issue in the event-listener repo whenever a change like this is made?

Whytecrowe committed:
change the way refunds work in SubRegistrar:

- add check for every revoke that gets a stakedAmount from StakePayment and proceeds to refund if it's > 0
- add state var for StakePayment, add setter, add event
- adapt deploy helpers

Whytecrowe committed:
add temporary test to validate the logic
